library/alloc/src/collections/binary_heap.rs

```diff
@@ -1568,20 +1568,36 @@ impl<T: Ord, I: IntoIterator<Item = T>> SpecExtend<I> for BinaryHeap<T> {
     }
 }
 
+impl<T: Ord> SpecExtend<Vec<T>> for BinaryHeap<T> {
```

Comment:
This shouldn't be necessary if `extend_desugared` now delegates to `vec.extend` which already has specializations to cover this case.

Reply:
Arguably it could be optimized to append from `self` to `other` if `other` has more elements since this would copy less data while still needing a full rebuild anyway. This is not the case of the current implementation though.